packages/terra-worklist-data-grid/src/subcomponents/Cell.jsx

```diff
@@ -0,0 +1,157 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import * as KeyCode from 'keycode-js';
+import '../_elementPolyfill';
+import { injectIntl } from 'react-intl';
+import classNames from 'classnames/bind';
+import ThemeContext from 'terra-theme-context';
+import WorklistDataGridUtils from '../utils/WorklistDataGridUtils';
+import styles from './Cell.module.scss';
+import cellShape from '../proptypes/cellShape';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * String identifier of the row in which the Cell will be rendered.
+   */
+  rowId: PropTypes.string.isRequired,
+  /**
+   * String identifier of the column in which the Cell will be rendered.
+   */
+  columnId: PropTypes.string.isRequired,
+  /**
+   * The cell's row position in the grid. This is zero based.
+   */
+  rowIndex: PropTypes.number,
+  /**
+   * The cell's column position in the grid. This is zero based.
+   */
+  columnIndex: PropTypes.number,
+  /**
+   * The information about the cell.
+   */
+  cell: cellShape.isRequired,
```

Comment:
Can we split this out into its contents: `content`, `isMasked`, and `isSelectable`? Those seem like they should be props on the Cell component rather than encompassed by a `cell` prop.

That will also make it easier to tell how `content`, `isMasked`, and `isSelectable` are defined via the Cell propTypes/props table.

Reply:
This is how data is passed into the WorklistDataGrid from consumers.  We would want to have a shape for the props being passed.

Reply:
Isn't the `Cell` component used internally? (Example: in [Row.jsx](https://github.com/cerner/terra-framework/pull/1676/files#diff-542c04ec82fe30d4fd6235a7ac912256c6469f2f05a311238bc20ee1bd5ac9e4R94-R105)) We would just structure it as:

```js
      <Cell
        rowId={rowId}
        columnId={columnId}
        rowIndex={cellRowIndex}
        columnIndex={cellColumnIndex}
        key={`${rowId}_${columnId}`}
        isTabStop={tabStopColumnIndex === cellColumnIndex}
        isSelected={!hasSelectableRows && selectedCellColumnId === columnId}
        isMasked={cell.isMasked}
        isSelectable={cell.isSelectable}
        content={cell.content}
        isRowHeader={isRowHeader}
        onCellSelect={handleCellSelect}
      />
```

`Cell` does not necessarily need to follow the prop structure at the `WorklistDataGrid` level. The reason is mainly readability in the `Cell.jsx` file.

Reply:
Oh, I see now what you mean about the prop validation for the consumer-provided cell data. Still, it seems odd to me to pass the data through 3+ layers of components before doing the validation.

Reply:
Let's discuss and we can adjust based on outcome.

Reply:
Changed based on discussion